<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find human_robot_collaboration)/rosconsole.conf"/>
    <!-- <include file="$(find human_robot_collaboration)/launch/baxter_marker_publisher.launch" /> -->
    <include file="$(find svox_tts)/launch/svox_tts.launch" />

    <arg name="forceThresholdL"      default="2.0"/>    <!-- [N] -->
    <arg name="forceThresholdR"      default="2.0"/>    <!-- [N] -->
    <arg name="forceFilterVarianceL" default="0.001"/>
    <arg name="forceFilterVarianceR" default="1.0"/>
    <arg name="relativeForceThresholdL"   default="65.0"/>
    <arg name="relativeForceThresholdR"   default="1.0"/>
    <arg name="speechPopUpDuration"  default="8.0"/>    <!-- [s] -->

    <!-- If the robot will recover from an error or will wait the external planner's decisions -->
    <arg name="internalRecovery"    default="false"/>
    <!-- If the system should use the robot or not (useful for debugging) -->
    <arg name="useRobot"            default="true"/>

    <param name="action_provider/force_threshold_left"  value="$(arg forceThresholdL)"/>
    <param name="action_provider/force_threshold_right" value="$(arg forceThresholdR)"/>
    <param name="action_provider/force_filter_variance_left"        value="$(arg forceFilterVarianceL)"/>
    <param name="action_provider/force_filter_variance_right"       value="$(arg forceFilterVarianceR)"/>
    <param name="action_provider/relative_force_threshold_left"     value="$(arg relativeForceThresholdL)"/>
    <param name="action_provider/relative_force_threshold_right"    value="$(arg relativeForceThresholdR)"/>

    <param name="action_provider/internal_recovery"     value="$(arg internalRecovery)"/>
    <param name="action_provider/use_robot"             value="$(arg useRobot)"/>
    <param name="baxter_display/speech_duration"        value="$(arg speechPopUpDuration)"/>

    <!-- Objects database for the left arm -->
    <rosparam param = "action_provider/objects_left">
        "table_top"   : 200
        "leg_1"       : 150
        "leg_2"       : 151
        "leg_3"       : 152
        "leg_4"       : 153
    </rosparam>

    <!-- Objects database for the right arm -->
    <rosparam param = "action_provider/objects_right">
        "screwdriver"  : 1
        "screws_box"   : 2
        "brackets_box" : 3
    </rosparam>

    <!-- <node pkg="human_robot_collaboration" type="baxter_display"  name="baxter_display"  output="screen"/> -->
    <!-- <node pkg="human_robot_collaboration" type="modular_action_provider" name="modular_action_provider" output="screen"/> -->


    <node name="image_view_hsv" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result"/>
        <param name="autosize" value="true" />
    </node>

    <param name="hsv_detector/reference_frame" value="/base"/>
    <param name="hsv_detector/camera_frame"    value="/right_hand_camera"/>
    <param name="hsv_detector/area_threshold"  value="250"/>

    <node pkg="human_robot_collaboration" type="hsv_detector" name="hsv_detector" output="screen" respawn="true">
        <remap from="hsv_detector/image" to="/cameras/right_hand_camera/image"/>
        <remap from="hsv_detector/camera_info" to="/cameras/right_hand_camera/camera_info"/>
    </node>

    <rosparam param = "hsv_detector/objects_db">
        "screwdriver" : { size: [0.099, 0.030], HSV: { H: [160,  10], S: [ 70, 166], V: [10,  66]}}
         <!-- This should be the blue box -->
        "screws_box"  : { size: [0.111, 0.111], HSV: { H: [ 80, 130], S: [120, 216], V: [70, 146]}}
         <!-- This should be the green box -->
        "brackets_box": { size: [0.111, 0.111], HSV: { H: [ 30, 100], S: [ 80, 246], V: [20,  96]}}
    </rosparam>

    <!-- <node name="image_baxter_display" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/robot/xdisplay"/>
        <param name="autosize" value="true" />
    </node> -->

    <!-- <node pkg="human_robot_collaboration" type="remote_control_node" name="l_srv" output="screen" args="left" />
    <node pkg="human_robot_collaboration" type="remote_control_node" name="r_srv" output="screen" args="right" /> -->

    <!-- <node name="image_saver" pkg="image_view" type="image_saver" respawn="false" output="log">
        <remap from="image" to="/robot/xdisplay"/>
        <param name="autosize" value="true" />
    </node> -->

    <!-- <node name="video_recorder" pkg="image_view" type="video_recorder" respawn="false" output="log">
        <remap from="image" to="/aruco_marker_publisher/result"/>
        <param name="fps" value="30" />
    </node> -->

</launch>
